src/sentry/search/events/builder.py

```diff
@@ -2774,6 +2776,66 @@ def get_snql_query(self) -> List[Request]:
         return queries
 
     def run_query(self, referrer: str, use_cache: bool = False) -> Any:
+        if self.use_metrics_layer:
+            from sentry.snuba.metrics.datasource import get_series
+            from sentry.snuba.metrics.mqb_query_transformer import (
+                tranform_mqb_query_to_metrics_query,
+            )
+
+            if self.is_performance:
+                use_case_id = UseCaseKey.PERFORMANCE
+            else:
+                use_case_id = UseCaseKey.RELEASE_HEALTH
+            prefix = "generic_" if self.dataset is Dataset.PerformanceMetrics else ""
+
+            snuba_query = Query(
+                match=Entity(f"{prefix}metrics_distributions", sample=self.sample_rate),
+                # Metrics doesn't support columns in the select, and instead expects them in the groupby
+                select=self.aggregates
+                + [
+                    # Team key transaction is a special case sigh
+                    col
+                    for col in self.columns
+                    if isinstance(col, Function) and col.function == "team_key_transaction"
+                ],
+                array_join=self.array_join,
+                where=self.where,
+                having=self.having,
+                groupby=self.groupby,
+                orderby=[],
+                granularity=self.granularity,
+            )
+            try:
+                metric_query = tranform_mqb_query_to_metrics_query(snuba_query)
+                metrics_data = get_series(
+                    projects=self.projects,
+                    metrics_query=metric_query,
+                    use_case_id=use_case_id,
+                    include_meta=True,
+                )
+            except Exception as err:
+                raise IncompatibleMetricsQuery(err)
+            metric_layer_result: Any = {
+                "data": [],
+                "meta": metrics_data["meta"],
+            }
+            # metric layer adds bucketed time automatically but doesn't remove it
+            for meta in metric_layer_result["meta"]:
+                if meta["name"] == "bucketed_time":
+                    meta["name"] = "time"
+            for index, interval in enumerate(metrics_data["intervals"]):
+                # the metric layer changes the intervals to datetime objects when we want the isoformat
+                data = {self.time_alias: interval.isoformat()}
+                # only need the first thing in groups since we don't groupby
+                for key, value_list in (
```

Comment:
Is this TimeseriesMetricQueryBuilder expected to work for top-n/grouped results? And in that case do we need to handle those transformations here?

Reply:
Discussed that top-n isn't expected to be supported here.